internal/sets/match.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sets
+
+import (
+	"reflect"
+)
+
+type equalFunction func(e1, e2 interface{}) bool
+
+// Compare checks if the parameters are iterable and contains the same elements
+func Compare(listA, listB interface{}, eq equalFunction) bool {
+
+	a := reflect.ValueOf(listA)
+	ok := isIterable(a)
+	if !ok {
+		return false
+	}
+
+	b := reflect.ValueOf(listB)
+	ok = isIterable(b)
+	if !ok {
+		return false
+	}
+
+	if a.IsNil() != b.IsNil() {
+		return false
+	}
```

Comment:
when they are both nil, we can already short-circuit and return true, no?

Reply:
done